eclipse-scout-core/src/table/Table.ts

```diff
@@ -6156,8 +6466,90 @@ export class Table extends Widget implements TableModel, Filterable<TableRow> {
+
+      // Install a table listener that stores all changes into the GLOBAL profile.
+      // This is only done _after_ applying the initial state, so that no events were triggered.
+      uiPreferences.installTableListener(this);
+    } else {
+      // Uninstall listener
+      uiPreferences.uninstallTableListener(this);
+    }
+  }
+
+  setInitialUiPreferences(initialUiPreferences: TableClientUiPreferenceProfileDo) {
+    this._initialUiPreferences = initialUiPreferences;
+  }
+
+  get initialUiPreferences(): TableClientUiPreferenceProfileDo {
+    return this._initialUiPreferences;
+  }
+
+  /**
+   * Applies the {@link initialUiPreferences} and clears the stored {@link UiPreferences#TABLE_PREFERENCE_PROFILE_ID_GLOBAL} profile.
+   * If the initial ui preferences were not previously set, this method has no effect.
+   */
+  resetToInitialUiPreferences() {
+    if (!this._initialUiPreferences) {
+      return;
+    }
+    uiPreferences.applyTablePreferencesProfile(this, this._initialUiPreferences, {
+      applyUserFilters: true
+    });
+    uiPreferences.clearGlobalTablePreferenceProfile(this);
+  }
+
+  applyColumnPreferences(columnPreferences: TableColumnClientUiPreferenceDo[]) {
+    let columnPreferencesMap = new Map(arrays.ensure(columnPreferences).map(pref => [pref.columnId, pref]));
+    let newColumns = this.columns
+      .filter(c => !c.guiOnly)
+      .sort((c1, c2) => {
+        let viewIndex1 = columnPreferencesMap.get(c1.buildUuid())?.viewIndex ?? Infinity;
+        let viewIndex2 = columnPreferencesMap.get(c2.buildUuid())?.viewIndex ?? Infinity;
+        return viewIndex1 - viewIndex2;
+      })
+      .map(column => {
+        this._applyColumnPreferencesToColumn(columnPreferencesMap.get(column.buildUuid()), column);
+        return column;
+      });
+
+    this.setColumns(newColumns);
+  }
+
+  protected _applyColumnPreferencesToColumn(columnPreferences: TableColumnClientUiPreferenceDo, column: Column<any>) {
+    if (!columnPreferences || !column) {
+      return; // can happen if preferences are applied before the customizer is installed or if preferences contains obsolete data
+    }
+
+    // Use setter for 'visible' property because it is a multidimensional property
+    column.setVisible(columnPreferences.visible, false); // parameter 'false' skips call of onColumnVisibilityChanged()
+    // FIXME bsh [js-preferences] this triggers an unnecessary change. As a matter of fact, _renderViewport is called too many times anyway!
+    column.setWidth(columnPreferences.width);
+
+    // Properties without setter (changes will be applied later by _setColumns)
+    column.sortIndex = columnPreferences.sortOrder;
+    column.sortAscending = columnPreferences.sortAscending;
+    column.grouped = columnPreferences.groupingActive;
+
+    if (column instanceof NumberColumn) {
+      column.setAggregationFunction(columnPreferences.aggregationFunctionId as NumberColumnAggregationFunction);
```

Comment:
Maybe add comment that setter is required to update the aggregation and background effect functions (e..g column.aggrStart etc.)